egs2/wsj0-2mix/enh1/enh.sh

```diff
@@ -0,0 +1,682 @@
+                         for training a denoising model (default="${use_noise_ref}")
+
+    # Enhancement related
+    --inference_args      # Arguments for enhancement in the inference stage (default="${inference_args}")
+    --inference_enh_model # Enhancement model path for inference (default="${inference_enh_model}").
+
+    # Evaluation related
+    --scoring_protocol    # Metrics to be used for scoring (default="${scoring_protocol}")
+    --ref_channel         # Reference channel of the reference speech will be used if the model
+                            output is single-channel and reference speech is multi-channel
+                            (default="${ref_channel}")
+
+    # [Task dependent] Set the datadir name created by local/data.sh
+    --train_set     # Name of training set (required).
+    --dev_set       # Name of development set (required).
+    --eval_sets     # Names of evaluation sets (required).
+    --enh_speech_fold_length # fold_length for speech data during enhancement training  (default="${enh_speech_fold_length}").
+EOF
+)
+
+log "$0 $*"
+. utils/parse_options.sh
+
+if [ $# -ne 0 ]; then
+    log "${help_message}"
+    log "Error: No positional arguments are required."
+    exit 2
+fi
+
+. ./path.sh
+. ./cmd.sh
+
+
+# Check required arguments
+[ -z "${train_set}" ] && { log "${help_message}"; log "Error: --train_set is required"; exit 2; };
+[ -z "${dev_set}" ] &&   { log "${help_message}"; log "Error: --dev_set is required"  ; exit 2; };
+[ -z "${eval_sets}" ] && { log "${help_message}"; log "Error: --eval_sets is required"; exit 2; };
+
+# Check feature type
+if [ "${feats_type}" = raw ]; then
+    data_feats=${dumpdir}/raw
+elif [ "${feats_type}" = fbank_pitch ]; then
+    data_feats=${dumpdir}/fbank_pitch
+elif [ "${feats_type}" = fbank ]; then
+    data_feats=${dumpdir}/fbank
+elif [ "${feats_type}" == extracted ]; then
+    data_feats=${dumpdir}/extracted
+else
+    log "${help_message}"
+    log "Error: not supported: --feats_type ${feats_type}"
+    exit 2
+fi
+
+
+
+
+
+# Set tag for naming of model directory
+if [ -z "${enh_tag}" ]; then
+    if [ -n "${enh_config}" ]; then
+        enh_tag="$(basename "${enh_config}" .yaml)_${feats_type}"
+    else
+        enh_tag="train_${feats_type}"
+    fi
+    # Add overwritten arg's info
+    if [ -n "${enh_args}" ]; then
+        enh_tag+="$(echo "${enh_args}" | sed -e "s/--/\_/g" -e "s/[ |=]//g")"
+    fi
+fi
+
+
+# The directory used for collect-stats mode
+enh_stats_dir="${expdir}/enh_stats_${fs}"
+# The directory used for training commands
+enh_exp="${expdir}/enh_${enh_tag}"
+
+# ========================== Main stages start from here. ==========================
+
+if [ ${stage} -le 1 ] && [ ${stop_stage} -ge 1 ]; then
+    log "Stage 1: Data preparation for data/${train_set}, data/${dev_set}, etc."
+    # [Task dependent] Need to create data.sh for new corpus
+    local/data.sh ${local_data_opts}
+fi
+
+if [ ${stage} -le 2 ] && [ ${stop_stage} -ge 2 ]; then
```

Comment:
Do we need this?
(probably it is effective but I don't see speed perturbation is used for separation. I'm curious. If it is effective, we can keep it!)

Reply:
I'm also curious. Let me have a try : )